### DISCUSSION TITLE
Stop validation and return on first error

### DISCUSSION THREAD

**Discussion by Sajarin-M** [react-hook-form/react-hook-form#9740] [Q&A] [answered]
I was wondering if it is possible to stop validation and return on the first error with `react-hook-form`, similar to the `abortEarly` option in the JOI validation library. I want to use `react-hook-form` to validate a form, but I only want to display the first error to the user. Is there a way to do this in `react-hook-form`? Thank you for your time.

**Comment by Moshyfawn** [comment]
I'm not familiar with Joi's `abortEarly`, TBH.

RHF has the `useForm` `criteriaMode: "firstError | all"` prop you can set to only return the first field error.

You can always do `Object.values(errors).at(0)` to get the first error property as well.

**Reply by Sajarin-M** [reply]
Thank you 🙏 for your response. I understand that the `criteriaMode` prop can be set to `firstError` to return the first field error, but this does not stop validating the next field. I want RHF to stop validating all other fields if an error is encountered in any one field. 

To provide a specific example, let's say that we have a form with two fields: `email` and `password`. We want to use RHF to validate the form, but we only want to display one error to the user at a time. If the user leaves the `email` field empty and submits the form, we want RHF to stop validating the `password` field and only display the error message for the `email` field.

Is it possible to achieve this behavior with RHF? If so, how can I do this? 😊

**Comment by leapful** [comment, marked as answer]
@Sajarin-M @Moshyfawn 

I think the Joi resolver from @hookform/resolvers library already support `abortEarly` option.

Working demo: https://codesandbox.io/s/xenodochial-framework-dn8ksu?file=/src/App.tsx

**Reply by Sajarin-M** [reply]
**Thank you 🙏 for your helpful response! I have one more question, if it's not too much trouble:** Is it possible to achieve this behavior using the native rules of `RHF`, rather than using a `resolver`? 🤔

**Reply by leapful** [reply]
@Sajarin-M 

There is currently no option to abort early built-in validators from react-hook-form. It would be easier to write a custom resolver to manage the validation order.
